url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/robot/SOCRobotResetThread.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Java Settlers - An online multiplayer version of the game Settlers of Catan
+ * Copyright (C) 2003  Robert S. Thomas
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+ *
+ * The author of this program can be reached at thomas@infolab.northwestern.edu
+ **/
+package soc.robot;
+
+
+/**
+ * When this thread wakes up it calls the disconnectReconnect
+ * method of the SOCRobotClient that spawned it
+ *
+ * @author Robert S Thomas
+ */
+public class SOCRobotResetThread extends Thread
+{
+    SOCRobotClient client;
+    boolean sleeping;
+    boolean alive;
+
+    /**
+     * constructor
+     *
+     * @param cl   the robot client
+     */
+    public SOCRobotResetThread(SOCRobotClient cl)
+    {
+        client = cl;
+        alive = true;
+        sleeping = true;
+    }
+
+    /**
+     * DOCUMENT ME!
+     */
+    public void run()
+    {
+        while (alive)
+        {
+            //
+            //  start by sleeping so another
+            //  thread has a chance to put it to sleep
+            //
+            while (sleeping)
+            {
+                sleeping = false;
+
+                try
+                {
+                    sleep(300000);
+                }
+                catch (InterruptedException exc)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)